src/contracts/middleware/BLSPubkeyRegistry.sol

```diff
@@ -0,0 +1,211 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity =0.8.12;
+
+
+import "../interfaces/IBLSPubkeyRegistry.sol";
+import "../interfaces/IRegistryCoordinator.sol";
+import "../interfaces/IBLSPublicKeyCompendium.sol";
+
+import "../libraries/BN254.sol";
+
+import "forge-std/Test.sol";
+
+
+contract BLSPubkeyRegistry is IBLSPubkeyRegistry, Test {
+
+    bytes32 internal constant ZERO_PK_HASH = hex"ad3228b676f7d3cd4284a5443f17f1962b36e491b30a40b2405849e597ba5fb5";
+
+    BN254.G1Point internal _globalApk;
+
+    IRegistryCoordinator public registryCoordinator;
+    IBLSPublicKeyCompendium public immutable pubkeyCompendium;
+
+
+    ApkUpdate[] public globalApkUpdateList;
+
+    mapping(uint8 => ApkUpdate[]) public quorumToApkUpdates;
+    
+    mapping(uint8 => BN254.G1Point) public quorumToApk;
+
+    event RegistrationEvent(
+        address indexed operator,
+        bytes32 pubkeyHash,
+        bytes32 globalApkHash
+    );
+
+    event DeregistrationEvent(
+        address indexed operator,
+        bytes32 pubkeyHash,
+        bytes32 globalApkHash
+    );
+
+    modifier onlyRegistryCoordinator() {
+        require(msg.sender == address(registryCoordinator), "BLSPubkeyRegistry.onlyRegistryCoordinator: caller is not the registry coordinator");
+        _;
+    }
+
+
+    constructor(
+        IRegistryCoordinator _registryCoordinator,
+        IBLSPublicKeyCompendium _pubkeyCompendium
+    ){
+        registryCoordinator = _registryCoordinator;
+        pubkeyCompendium = _pubkeyCompendium;
+
+        _initializeApkUpdates();
+    }
+
+
+    function registerOperator(address operator, uint8[] memory quorumNumbers, BN254.G1Point memory pubkey) external onlyRegistryCoordinator returns(bytes32){
```

Comment:
comments from interfaces need to be brought over

Reply:
added